### PR TITLE
Cleanup deploy parameters

### PR DESCRIPTION
**Pull request by trieloff**
As discussed in #4

### CODE REVIEW

File: test/testDeployCmd.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    new CLI()
+      .run(['deploy',
+        '--wsk-auth', 'secret-key',
+        '--wsk-namespace', 'hlx',
+        '--no-auto',
+        '--dirty',
+        '--dry-run',
+        '--target', 'test/integration/.hlx/build',
+      ]);
+    done();
```
Comment (tripodsan) [inline test/testDeployCmd.js]: done should be invoked in the `then` of the run. and `.catch(done)` is missing.

File: test/testDeployCli.js [hunk 2]
```diff
@@ -68,42 +69,65 @@ describe('hlx deploy', () => {
...
+    sinon.assert.calledWith(mockDeploy.withLogglyAuth, '');
+    sinon.assert.calledWith(mockDeploy.withTarget, '.hlx/build');
+    sinon.assert.calledWith(mockDeploy.withDocker, 'trieloff/custom-ow-nodejs8:latest');
+    sinon.assert.calledWith(mockDeploy.withPrefix, 'git-github-com-example-project-helix--master--');
+    sinon.assert.calledWith(mockDeploy.withDefault, undefined);
+    sinon.assert.calledOnce(mockDeploy.run);
+  });
+
+  it('hlx deploy works with arguments provided in environment', () => {
+    process.env.HLX_WSK_AUTH = 'sekret-key';
```
Comment (tripodsan) [inline test/testDeployCli.js]: test all other possible arguments als well

File: test/testDeployCmd.js (hunk 1 above)
Comment (trieloff) [reply test/testDeployCmd.js]: `run()` does not return a promise, it returns the parsed `args`.

File: test/testDeployCli.js (hunk 2 above)
Comment (trieloff) [reply test/testDeployCli.js]: I think that would be overkill, as the `env()` method of `yargs` is well-tested and we won't get any more functional coverage: https://github.com/yargs/yargs/blob/6ee2c82df515cb1b062e4135a5dd9c386fed2b21/test/yargs.js#L756

**Pull request merged**